packages/flutter/test/widgets/media_query_test.dart

```diff
@@ -564,4 +564,102 @@ void main() {
     expect(outsideBoldTextOverride, false);
     expect(insideBoldTextOverride, true);
   });
+
+  testWidgets('MediaQueryData can not be initialized with null values', (WidgetTester tester) async {
+
+    String generateErrorMessage(String parameterName){
+      return '\'$parameterName != null\': is not true';
+    }
+
+    try{
+      MediaQueryData(size: null);
+    }on AssertionError catch(error){
+      expect(error.toString(), contains(generateErrorMessage('size')));
+    }
```

Comment:
Also, these asserts might have to get tested individually... Say that the assert never throws, the test will pass silently because there's nothing preventing it from failing. It has to be something like this:
```suggestion
    try {
      MediaQueryData(size: null);
    } on AssertionError catch (Exception error) {
      expect(error.toString(), contains(generateErrorMessage('size')));
      return; // to end the unit test
    }
    fail('The assert was never called when it should have been'); // this line causes the unit test to fail
```

To test if your unit test is working, remove the assert from the code and check if the test fails